test/Datadog.Trace.TestHelpers/HeadersCollectionTestBase.cs

```diff
@@ -29,5 +30,18 @@ public static IEnumerable<object[]> GetHeadersInvalidSamplingPrioritiesCartesian
                    from invalidSamplingPriority in HeadersCollectionTestHelpers.GetInvalidSamplingPriorities().SelectMany(i => i)
                    select new[] { header, invalidSamplingPriority };
         }
+
+        internal static void AssertExpected(IHeadersCollection headers, string key, string expected)
```

Comment:
You could make this public so you don't have to add an `InternalsVisibleTo` attribute. This won't be consumed by developers so I don't see any reason to keep this internal

Reply:
IHeadersCollection is also internal, that's the original reason.